### ISSUE TITLE
Can you talk about how to adjust these hyperparameters?

### ISSUE THREAD

**Issue by q-rong** [hobincar/SA-LSTM#8]
Hi,
I would like to ask you, how can I better adjust these hyperparameters? What's the trick?Or do we have to try them one by one?

Looking forward to your reply, thank you.

**Comment by hobincar** [comment]
Hi,

IMHO, there's no such oracle for adjusting the hyperparameters. In my experience, all of the hyperparameters including batch size, learning rate, the size of LSTM hidden states, the number of frames that you sample from the input video, etc. will affect the performance of your model.

One known practice in hyperparameter search is doing in [a random manner rather than a grid manner](http://www.jmlr.org/papers/volume13/bergstra12a/bergstra12a.pdf). Or for those who are tired of finetuning their hyperparameters, there are services from Google and MS that search hyperparameter spaces on behalf of you.
